components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/logout/LogoutRequestSender.java

```diff
@@ -140,14 +142,45 @@ public LogoutReqSenderTask(SingleLogoutRequestDTO logoutReqDTO) {
         public void run() {
             List<NameValuePair> logoutReqParams = new ArrayList<NameValuePair>();
             // set the logout request
-            logoutReqParams.add(new BasicNameValuePair("SAMLRequest", logoutReqDTO.getLogoutResponse()));
+            String startSoapBinding = "<SOAP-ENV:Envelope xmlns:SOAP-ENV=\"http://schemas.xmlsoap.org/soap/envelope/\">" +
+                    "<SOAP-ENV:Body>";
+            String endSoapBinding = "</SOAP-ENV:Body>" +
+                    "</SOAP-ENV:Envelope>";
+            String soapAction = "http://www.oasis-open.org/committees/security";
 
-            if (log.isDebugEnabled()) {
-                try {
-                    log.debug("SAMLRequest : " + SAMLSSOUtil.decodeForPost(logoutReqDTO.getLogoutResponse()));
-                } catch (IdentityException e) {
-                    log.debug("Error in decoding logout request.", e);
+            StringBuffer logoutRequestWithSoapBinding = new StringBuffer();
+            String decodedSAMLRequest = null;
+
+            boolean propertySAMLSOAPBindingEnabled = IdentityConfigParser.getInstance()
+                    .getConfiguration().containsKey(SAMLSSOConstants.SLO_SAML_SOAP_BINDING_ENABLED);
+            boolean isSAMLSOAPBindingEnabled;
+
+            if (propertySAMLSOAPBindingEnabled) {
+                isSAMLSOAPBindingEnabled = Boolean.parseBoolean(IdentityConfigParser.getInstance()
+                        .getConfiguration().get(SAMLSSOConstants.SLO_SAML_SOAP_BINDING_ENABLED).toString());
+            } else {
+                isSAMLSOAPBindingEnabled = false;
+            }
+
+            try {
+                decodedSAMLRequest = SAMLSSOUtil.decodeForPost(logoutReqDTO.getLogoutResponse());
```

Comment:
Send the SLO request inside the SingleLogoutRequestDTO without encoding and encode here as required.